src/main/scala/com/amazon/deequ/analyzers/jdbc/JdbcAnalyzer.scala

```diff
@@ -1,287 +0,0 @@
-/**
- * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License"). You may not
- * use this file except in compliance with the License. A copy of the License
- * is located at
- *
- *     http://aws.amazon.com/apache2.0/
- *
- * or in the "license" file accompanying this file. This file is distributed on
- * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
- * express or implied. See the License for the specific language governing
- * permissions and limitations under the License.
- *
- */
-
-package com.amazon.deequ.analyzers.jdbc
-
-import java.sql.{JDBCType, ResultSet}
-
-import com.amazon.deequ.analyzers.State
-import com.amazon.deequ.analyzers.runners._
-import com.amazon.deequ.metrics.{DoubleMetric, Entity, Metric}
-
-import java.sql.Types._
-
-import scala.util.{Failure, Success}
-
-trait JdbcAnalyzer[S <: State[_], +M <: Metric[_]] {
```

Comment:
Do you think we can utilize more of the already existing code in `Analyzer` and deequ `Preconditions` for example?

Reply:
Something like a parent class for both jdbc and spark analyzer?

Reply:
Yes, I was thinking that you can subclass `Analyzer` in `JdbcAnalyzer` for example.

Could you also please include the base classes you have already implemented in the PR? (couldn't find `JdbcScanShareableFrequencyBasedAnalyzer` here)

Reply:
I think that would be a good solution.

The `JdbcScanShareableFrequencyBasedAnalyzer` is in `GroupingAnalyzer.scala`.